ui/src/main/java/org/pentaho/di/ui/job/entries/trans/JobEntryBaseDialog.java

```diff
@@ -707,14 +707,16 @@ public void modifyText( ModifyEvent modifyEvent ) {
         } catch ( KettleException ignored ) {
           // Ignore errors
         }
-        display.asyncExec( () -> {
-          if ( (Boolean) items.get( IS_PENTAHO ) ) {
-            wWaitingToFinish.setSelection( false );
-            wWaitingToFinish.setEnabled( false );
-          } else {
-            wWaitingToFinish.setEnabled( true );
-          }
-        } );
+        if ( wWaitingToFinish != null && !wWaitingToFinish.isDisposed() ) {
```

Comment:
I think this check should be done inside the asyncExec. When we do an asyncExec it does not immediately execute. We are creating a new Runnable that will be run by the UI thread. 
So, in the point where we check the wWaitingToFinish everything might be ok and we create the new async execution. When the async execution actually runs then wWaitingToFinish might already be null or disposed.

Reply:
I understand that, and i also considered that option, but yhe question that came into my mind was, why even create a running if the object that is going to affected is already disposed?

Reply:
New commit added. Please consider squashing it when merging.